src/function.rs

```diff
@@ -372,6 +375,30 @@ fn path_exists(evaluator: &Evaluator, path: &str) -> Result<String, String> {
   )
 }
 
+fn pick(_evaluator: &Evaluator, count: &str, alphabet: Option<&str>) -> Result<String, String> {
+  let alphabet = alphabet.unwrap_or("0123456789abcdef");
+  if alphabet.is_empty() {
+    return Err(String::from("no characters in alphabet"));
+  }
+  let mut alphabet_unique = HashSet::<char>::with_capacity(alphabet.len());
```

Comment:
I think you can do a `HashSet::from_iter` here! It should be something like:

```rust
HashSet::<char>::from_iter(alphabet.chars())
```

I wonder if it's worth returning an error when we know we'll get the repeated characters (I think ya'll talked about this as well, so I'm definitely fine either way)

Reply:
Ah so this kind of touches on the second point -- do we actually need to tell people which character was repeated? If we say *repeated characters will be de-duplicated* or something to that effect in the doc, I don't think people would mind -- usually `"aabcdef..."` is a bug anyway.

Reply:
Yup, makes sense to me!